### ISSUE TITLE
the typing the second parameter of push.apply should be readonly T[], not T[]

### ISSUE THREAD

**Issue by LongTengDao** [microsoft/TypeScript#32276]
**TypeScript Version:**  3.4.1

**Search Terms:**

**Code**

```ts
function main (input :readonly any[]) {
    const output :any[] = [];
    Array.prototype.push.apply(output, input);
    return output;
}
```

**Expected behavior:**

Not to report an error.

**Actual behavior:**

Report error `TS2345`.

**Playground Link:** 

**Related Issues:**

**Comment by MartinJohns** [comment]
In TypeScript 3.5.2 the second parameter is `readonly any[]`, if you have the compiler option `strictBindCallApply` enabled. This option was released with TypeScript 3.2 (November 2018).

#27028

**Comment by vbraun** [comment]
This is still broken on Typescript 4.0.2, and has been since at least 3.5.1 (oldest version on the typescript playground):

https://www.typescriptlang.org/play?ts=4.0.2#code/GYVwdgxgLglg9mABAWwIYyQCgwBxFRALgCcBTVAEwQBsBPRVMWgbQF0BKRAbwChF+B-CAgDOBOPjwFCjFq0QBeRGwDcfQQICCxYqloA6HMThQTtHKUMgRAC32ocOOpglQpAGkS587NRoFkUCDESK5SfgIAvkA
